packages/data-migration/src/MigrationRunner.ts

```diff
@@ -196,10 +238,16 @@ export class MigrationRunner {
             ids.add(id);
         }
     }
+
+    private async createCheckpoint(migration: DataMigration, checkpoint: unknown) {
```

Comment:
Maybe `createCheckpoint` can throw some `MigrationCheckpointFailed` error if it fails.
It could contain the JSON.stringified data, which user can use to continue migration manually?